corehq/pillows/synclog.py

```diff
@@ -59,32 +62,49 @@ def process_change(self, change):
         if not synclog:
             return
 
-        version = None
-        app_id = None
+        user_id = synclog.get('user_id')
+        domain = synclog.get('domain')
+
+        if not user_id or not domain:
+            return
+
         try:
             sync_date = string_to_utc_datetime(synclog.get('date'))
         except (ValueError, AttributeError):
             return
+
         build_id = synclog.get('build_id')
-        if build_id:
-            version, app_id = get_version_and_app_from_build_id(synclog.get('domain'), build_id)
-        user_id = synclog.get('user_id')
+        device_id = synclog.get('device_id')
+        app_id = synclog.get('app_id')
+
+        if settings.USER_REPORTING_METADATA_BATCH_ENABLED:
+            UserReportingMetadataStaging.add_sync(domain, user_id, app_id, build_id, sync_date, device_id)
+        else:
+            mark_last_synclog(domain, user_id, app_id, build_id, sync_date, device_id)
+
+
+def mark_last_synclog(domain, user_id, app_id, build_id, sync_date, device_id):
+    user = CouchUser.get_by_user_id(user_id)
+    if not user:
+        return
+
+    version = None
+    if build_id:
+        version = get_version_from_build_id(domain, build_id)
+
+    save = update_last_sync(user, app_id, sync_date, version)
+    if version:
+        save |= update_latest_builds(user, app_id, sync_date, version)
+
+    app_meta = None
+    device_id = device_id
```

Comment:
This looks like you maybe left it here after a find and replace or something

Reply:
https://github.com/dimagi/commcare-hq/pull/25726/commits/a5f163f18679380efe03fa5fcfe89470e25e028d